api/base/pagination.py

```diff
@@ -236,6 +236,13 @@ def get_resource(self, kwargs):
         return Preprint.load(resource_id)
 
 
+class DraftRegistrationContributorPagination(NodeContributorPagination):
+
+    def get_resource(self, kwargs):
+        resource_id = kwargs.get('draft_id')
+        return DraftRegistration.load(resource_id)
```

Comment:
[`load`](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf/models/base.py#L88-L97) just wraps `DoesNotExist` error handling, and optionally allows for `SELECT FOR UPDATE` row locking. Is perfectly fine for use.

Reply:
i guess i have a bias for using well-maintained and documented third-party APIs -- seems nicer to directly use django's `select_for_update` when we need it, and explicitly handle errors rather than swallow and return `None` (much easier to trace a `DoesNotExist` than an `AttributeError`)